core/src/main/scala/scala/collection/generic/Signalling.scala

```diff
@@ -131,19 +131,21 @@ trait AtomicIndexFlag extends Signalling {
   abstract override def setIndexFlag(f: Int) = intflag.set(f)
   abstract override def setIndexFlagIfGreater(f: Int) = {
     var loop = true
-    do {
+    while({
```

Comment:
can just be `while (loop) { ... }` since `loop` starts out as `true`. this looks significantly cleaner IMO (though they are of course equivalent)

Reply:
That's what I did...